### PR TITLE
On 1763 multiple answer questions save and continue

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**
Save and continue is now possible for multiple answer pages e.g. Directors.

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Application/Apply/UpdatePageAnswers/UpdatePageAnswersHandler.cs [hunk 1]
```diff
@@ -4,285 +4,251 @@
...
-        if (answer != null)
+        private static bool ExistingUpload(string questionId, List<PageOfAnswers> pagePageOfAnswers)
         {
-          pageAnswers.Answers.Add(answer);
+            return pagePageOfAnswers.Any(poa => poa.Answers.Any(a => a.QuestionId == questionId && a.Value != ""));
         }
-      }
-
-      return validationPassed;
     }
```
Comment (davidgouge) [inline src/SFA.DAS.ApplyService.Application/Apply/UpdatePageAnswers/UpdatePageAnswersHandler.cs]: Not entirely sure what happened with this file when editing, but it makes it extremely hard to see what's changed.

File: src/SFA.DAS.ApplyService.Web/Controllers/ApplicationController.cs [hunk 2]
```diff
@@ -304,92 +304,30 @@ private void ProcessPageVmQuestionsForStandardName(List<QuestionViewModel> pageV
...
+        /// <summary>
+        /// Checks if answers must be validated
+        /// </summary>
+        /// <param name="applicationId">The application identifier.</param>
+        /// <param name="sequenceId">The sequence identifier.</param>
+        /// <param name="sectionId">The section identifier.</param>
+        /// <param name="pageId">The page identifier.</param>
+        /// <param name="answers">The answers.</param>
+        /// <param name="inputEnteredRegex">The input entered regex which determines if an answer has been entered.</param>
+        /// <returns>True when any answer has been entered or no previous answers have been entered for a multi answer page; False otherwise.</returns>
```
Comment (davidgouge) [inline src/SFA.DAS.ApplyService.Web/Controllers/ApplicationController.cs]: The majority of this summary doesn't tell us any more than the method signature and is just noise.

File: src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Controls/_Date.cshtml [hunk 3]
```diff
@@ -14,6 +14,10 @@
     }
 }
 
+@* 
+    when there are multiple input elements with the same name the form variable on post back is (name = 'value1,value2,value3') this can be used to separate the components of the answer,
+    in the date component case since a ',' cannot be entered into any input element, this is a safe separator which can be used to separate the single answer value into 3 input elements.
+*@
```
Comment (davidgouge) [inline src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Controls/_Date.cshtml]: You can remove this comment.

File: src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Index.cshtml [hunk 4]
```diff
@@ -83,7 +83,8 @@
                     </div>
                 }
 
-                <form method="post" asp-action="SaveAnswers" asp-controller="Application" novalidate enctype="multipart/form-data">
+                @* no action is specified for the form, post back to the get url to preserve the page hierarchy parameters *@
+                <form method="post" novalidate enctype="multipart/form-data">
```
Comment (davidgouge) [inline src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Index.cshtml]: And this one.

File: src/SFA.DAS.ApplyService.Web/Views/Application/Pages/MultipleAnswers.cshtml [hunk 5]
```diff
@@ -117,12 +117,10 @@
         </div>
     </div>
     
-
-
     <div class="govuk-grid-row">
         <div class="govuk-grid-column-two-thirds">
-
-            <form method="post" asp-action="SaveAnswersMulti" asp-controller="Application" novalidate enctype="multipart/form-data">
+            @* no action is specified for the form, post back to the get url to preserve the page hierarchy parameters *@
```
Comment (davidgouge) [inline src/SFA.DAS.ApplyService.Web/Views/Application/Pages/MultipleAnswers.cshtml]: And this one

**Comment by davidgouge** [comment]
Also seems to be tests missing for the new UpdatePageAnswersHandler changes at least. 👍

File: src/SFA.DAS.ApplyService.Application/Apply/UpdatePageAnswers/UpdatePageAnswersHandler.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ApplyService.Application/Apply/UpdatePageAnswers/UpdatePageAnswersHandler.cs]: This file was formatted with 2 space tabs; and all the others in the project with 4 space tabs so I reformatted the file - this worked fine in visual studio and the differ showed only the changes but it has totally thrown github - I'll try and resolve this

File: src/SFA.DAS.ApplyService.Web/Controllers/ApplicationController.cs (hunk 2 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ApplyService.Web/Controllers/ApplicationController.cs]: I'll rework that

File: src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Controls/_Date.cshtml (hunk 3 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Controls/_Date.cshtml]: That was part of some other changes I rolled back

File: src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Index.cshtml (hunk 4 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ApplyService.Web/Views/Application/Pages/Index.cshtml]: This is necessary comment as what this does is unexpected, post back to get url is a non-typical form post style

File: src/SFA.DAS.ApplyService.Application/Apply/UpdatePageAnswers/UpdatePageAnswersHandler.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ApplyService.Application/Apply/UpdatePageAnswers/UpdatePageAnswersHandler.cs]: Using the Diff settings drop down select 'Hide whitespace changes' - there is still some oddities as line move inwards to 4 space tabs - but this is acceptable given the file has been reformatted to match the others in the project.

**Pull request merged**